BLE_SM/source/main.cpp

```diff
@@ -459,35 +461,40 @@ class SMDeviceCentral : public SMDevice {
         }
     }
 
-    /** This is called by Gap to notify the application we connected,
-     *  in our case it immediately request pairing */
-    virtual void onConnectionComplete(const ble::ConnectionCompleteEvent &event)
-    {
-        if (event.getStatus() == BLE_ERROR_NONE) {
-            /* store the handle for future Security Manager requests */
-            _handle = event.getConnectionHandle();
+    void requestPairing() {
 
-            printf("Connected\r\n");
+        /* in this example the local device is the master so we request pairing */
+        ble_error_t error = _ble.securityManager().requestPairing(_handle);
 
-            /* in this example the local device is the master so we request pairing */
-            ble_error_t error = _ble.securityManager().requestPairing(_handle);
+         if (error) {
+             printf("Error during SM::requestPairing %d\r\n", error);
+             return;
+         }
 
-             if (error) {
-                 printf("Error during SM::requestPairing %d\r\n", error);
-                 return;
-             }
+        /* upon pairing success the application will disconnect */
 
-            /* upon pairing success the application will disconnect */
-        }
+		/* failed to connect - restart scan */
```

Comment:
this should be in onConnectionComplete

Reply:
What specifically should be in onConnectionComplete? I only see comments in the selected lines.

requestPairing is a call deferred to the event queue in onConnectionComplete.

Reply:
I meant that scanning should be restarted in onCOnnectionComplete in case request pairing is not called so that scanning is always resumed